### ISSUE TITLE
SNS subscribe to topic throws 502 when invoked via Java AWS client 

### ISSUE THREAD

**Issue by justinjose28** [localstack/localstack#1956]
While trying to subscribe to a SNS topic via Java AWS client, localstack throws 502 error. Even though exception is thrown, it is actually creating the subscription. Please find below the stacktrace I received for the issue

>  2020-01-16T14:33:26:ERROR:localstack.services.generic_proxy: Error forwarding request: 'arn:aws:sns:us-east-1:000000000000:proxy-local' Traceback (most recent call last):
localstack_demo |   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 300, in forward
localstack_demo |     updated_response = self.proxy.update_listener.return_response(**kwargs)
localstack_demo |   File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 158, in return_response
localstack_demo |     filter_policy
localstack_demo |   File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 263, in do_subscribe
localstack_demo |     SNS_SUBSCRIPTIONS[topic_arn].append(subscription)
localstack_demo | KeyError: 'arn:aws:sns:us-east-1:000000000000:proxy-local'

I tried with version 0.9.1 and it is working fine. 

Basically, this issue happen in latest versions when request is sent as a POST and the parameters/commands are sent as request body(as opposed to query parameters) which is what the Java AWS client does. If commands are sent as query parameters using GET/POST, it works fine.

**Comment by Michael-S-Smith** [comment]
I am struggling with what seems to be a very similar issue.
I tried for some time to get a locally-running spring boot app to subscribe to and SNS topic on localstack docker - constantly failed with 'Connection Refused' errors. I guessed this was some docker/networking issue, so I installed localstack directly on my Windows box.
I can create an SNS topic and successfully send it a message using aws cli. But after my app attempts to subscribe to the topic, any subsequent attempts to send it a message results in a 502. Localstack does recognize there is a subscription.

The code runs fine when pointing to a real AWS account.

I would like to try with localstack version 0.9.1 but I cant find that.

Below is the AWS CLI output showing
1) Successful publish to the topic
2) Bad gateway error after my app registers as a subscriber
3) list-subscriptions showing localstack recogizes the subscriber

Thank you for any help you can provide

**Comment by whummer** [comment]
Thanks for reporting @justinjose28 @Michael-S-Smith .

@justinjose28 How exactly are you constructing or receiving the topic ARN `arn:aws:sns:us-east-1:000000000000:proxy-local` - is it returned from the `CreateTopic` or `ListTopics` API calls?

Can you please post the output of this command, right after the error occurs? (assuming you have `awslocal` installed)
```
awslocal sns list-topics
```

@Michael-S-Smith Can you please also include the CLI command for the subscribe operation, as well as any errors logs in the LocalStack container with `DEBUG=1` enabled. Thanks

**Comment by thil1212** [comment]
@Michael-S-Smith 
 I am facing same problem have you resolved?

**Comment by Michael-S-Smith** [comment]
No sorry I gave up and took another route to implement what I needed.

**Comment by whummer** [comment]
Thanks again for providing the details. We were unable to reproduce this issue - it is likely already fixed in the `latest` version. An integration test has been added in #2406 to cover this functionality. If the problem persists, can you please post the detailed error logs (`DEBUG=1`) and provide a self-contained examples to reproduce this issue? Thanks!
